sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/ContainerRegistryBlobClient.cs

```diff
@@ -0,0 +1,503 @@
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Download a blob that is part of an artifact.
+        /// </summary>
+        /// <param name="digest">The digest of the blob to download.</param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns></returns>
+        public virtual async Task<Response<DownloadBlobResult>> DownloadBlobAsync(string digest, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(digest, "digest");
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ContainerRegistryBlobClient)}.{nameof(DownloadBlob)}");
+            scope.Start();
+            try
+            {
+                ResponseWithHeaders<Stream, ContainerRegistryBlobGetBlobHeaders> blobResult = await _blobRestClient.GetBlobAsync(_repositoryName, digest, cancellationToken).ConfigureAwait(false);
+
+                if (!ValidateDigest(blobResult.Value, digest))
+                {
+                    throw _clientDiagnostics.CreateRequestFailedException(blobResult.GetRawResponse(), "The manifest's digest according to the registry does not match the locally computed digest value.");
+                }
+
+                return Response.FromValue(new DownloadBlobResult(digest, blobResult.Value), blobResult.GetRawResponse());
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Delete a blob. The blob will only be removed from the repository if no other artifacts reference it.
+        /// </summary>
+        /// <param name="digest">The digest of the blob to delete.</param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns></returns>
+        public virtual Response DeleteBlob(string digest, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(digest, "digest");
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ContainerRegistryBlobClient)}.{nameof(DeleteBlob)}");
+            scope.Start();
+            try
+            {
+                ResponseWithHeaders<Stream, ContainerRegistryBlobDeleteBlobHeaders> blobResult = _blobRestClient.DeleteBlob(_repositoryName, digest, cancellationToken);
+                return blobResult.GetRawResponse();
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Delete a blob. The blob will only be removed from the repository if no other artifacts reference it.
```

Comment:
What is the error experience if another artifact references the blob?

Reply:
The service does ref-counting on the blobs, so they'd get a 202 from the delete, I believe, but the blob wouldn't actually be deleted until there are zero references to it.  (Even then the service only soft-deletes it b/c they had gotten so many customer issues with accidental deletes.)